src/services/outliningElementsCollector.ts

```diff
@@ -84,6 +85,7 @@ namespace ts.OutliningElementsCollector {
         let depth = 0;
         const maxDepth = 20;
         function walk(n: Node): void {
+            cancellationToken.throwIfCancellationRequested();

```

Comment:
We'll need to measure the cost here. Seems like a really heavily hit code path to add extra checks in.